mayday/file_test.go

```diff
@@ -0,0 +1,35 @@
+package mayday_test
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/coreos/mayday/mayday"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNonexistentFile(t *testing.T) {
+	file := mayday.File{Path: "/etc/nonexistent"}
+	err := file.Collect(workspace)
+	assert.Equal(t, err.Error(), `error opening source file: open /etc/nonexistent: no such file or directory`)
+}
+
+func TestFile(t *testing.T) {
+	file := mayday.File{Path: "/proc/meminfo"}
+	res := file.Collect(workspace)
+	assert.Nil(t, res)
+
+	f, err := ioutil.ReadFile(workspace + "/proc/meminfo")
+	assert.Nil(t, err)
+	assert.Contains(t, string(f), "MemTotal")
+}
+
+func TestFileWithLink(t *testing.T) {
+	file := mayday.File{Path: "/etc/crontab", Link: "cronfig"}
```

Comment:
This might cause a problem as many systems don't ship with the file /etc/crontab (including CoreOS by default):

```
$ ls -l /etc/crontab     
ls: cannot access /etc/crontab: No such file or directory
```
Possibly using `/etc/passwd` which should always exist in one form or another or similar to `/proc/meminfo` using `/proc/cpuinfo`?

Another possibility for these would be to create a file in `/tmp`, and then guaranteeing you can pull it in.  Then it's not even dependent on `procfs` being available.